### PR TITLE
490 project modalities list

### PR DESCRIPTION
**Pull request by davidsmejia**
## Issue Number

#490 

## Purpose/Implementation Notes

- Changes `Project.modalities` to `ArrayField` from `TextField`
- Adds migration for above change (supports revert)
- Updates filter options endpoint to handle this change
- Updates tests to handle this change

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

N/A

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Screenshots

n/a

### CODE REVIEW

**Review by arkid15r — APPROVED** [review]
LGTM

File: api/scpca_portal/models/project.py [hunk 1]
```diff
@@ -42,7 +43,7 @@ class Meta:
     has_spatial_data = models.BooleanField(default=False)
     human_readable_pi_name = models.TextField()
     includes_anndata = models.BooleanField(default=False)
-    modalities = models.TextField(blank=True, null=True)
+    modalities = ArrayField(models.TextField(), default=list, blank=True)
```
[the remark this answers is not in the document]
Comment (davidsmejia) [reply api/scpca_portal/models/project.py:46]: This was something I was actually curious about.

If we do not add `blank=True` for modalities then the `serializers.test_project` tests fail with the following error:
```bash
test_serializer_with_valid_data (scpca_portal.test.serializers.test_project.TestProjectSerializer) ... {'modalities': [ErrorDetail(string='This list may not be empty.', code='empty')]}
```  

I agree that not specifying `blank=True` while having a default value of `list` should be sufficient. However I was unable to get the tests to agree with me.

[the remark this answers is not in the document]
Comment (arkid15r) [reply api/scpca_portal/models/project.py:46]: It looks like DRF related issue. Try adding non-empty modalities field to `test::factories::LeafProjectFactory`, e.g. `modalities = ["CITE-seq"]`

[the remark this answers is not in the document]
Comment (davidsmejia) [reply api/scpca_portal/models/project.py]: yeah that was it, updated

**Pull request merged**